tests/performance/CatalogClientTest.cpp

```diff
@@ -0,0 +1,170 @@
+  }
+
+  std::shared_ptr<olp::client::OlpClientSettings>
+  CreateCatalogClientSettings() {
+    auto parameter = GetParam();
+
+    std::shared_ptr<olp::thread::TaskScheduler> task_scheduler = nullptr;
+    if (parameter.task_scheduler_capacity != 0) {
+      task_scheduler =
+          olp::client::OlpClientSettingsFactory::CreateDefaultTaskScheduler(
+              parameter.task_scheduler_capacity);
+    }
+    olp::authentication::Settings settings;
+    settings.task_scheduler = task_scheduler;
+    settings.network_request_handler = s_network;
+
+    olp::client::AuthenticationSettings auth_settings;
+    auth_settings.provider = []() { return "invalid"; };
+
+    auto client_settings = std::make_shared<olp::client::OlpClientSettings>();
+    client_settings->authentication_settings = auth_settings;
+    client_settings->task_scheduler = task_scheduler;
+    client_settings->network_request_handler = s_network;
+    client_settings->proxy_settings = GetLocalhostProxySettings();
+
+    return client_settings;
+  }
+
+ protected:
+  static std::shared_ptr<olp::http::Network> s_network;
+};
+
+std::shared_ptr<olp::http::Network> CatalogClientTest::s_network;
+
+namespace {
+
+void TestClientThread(
+    std::shared_ptr<olp::dataservice::read::CatalogClient> service_client,
+    std::chrono::duration<double, std::milli> sleep_interval, int runtime) {
+  auto start = std::chrono::steady_clock::now();
+  int partition_id = 0;
+  while (true) {
+    partition_id++;
+
+    auto request = olp::dataservice::read::DataRequest()
+                       .WithLayerId(kLayerId)
+                       .WithPartitionId(std::to_string(partition_id))
+                       .WithBillingTag(boost::none);
+
+    service_client->GetData(
+        request, [](olp::dataservice::read::DataResponse response) {});
+
+    std::this_thread::sleep_for(sleep_interval);
+
+    auto end = std::chrono::steady_clock::now();
+    std::chrono::duration<double, std::milli> elapsed = end - start;
+    if (elapsed.count() > runtime * 1000) {
+      return;
+    }
+  }
+}
+}  // namespace
+
+TEST_P(CatalogClientTest, ReadPartitions) {
+  auto parameter = GetParam();
+  std::chrono::duration<double, std::milli> sleep_interval(
+      1000. / parameter.requests_per_second);
+
+  auto olp_client_settings = CreateCatalogClientSettings();
+  olp::client::HRN hrn("hrn:here:data:::testhrn");
+
+  std::vector<std::thread> client_threads;
+
+  for (int client_id = 0; client_id < parameter.calling_thread_count;
+       client_id++) {
+    auto service_client =
+        std::make_shared<olp::dataservice::read::CatalogClient>(
+            hrn, olp_client_settings);
+    auto thread = std::thread(TestClientThread, std::move(service_client),
+                              sleep_interval, parameter.runtime);
+    client_threads.push_back(std::move(thread));
+  }
+
+  for (auto& thread : client_threads) {
+    thread.join();
+  }
+}
+
+//   int requests_per_second;
```

Comment:
Better to place these comments to the left of actual values (see lines 169 - 170) in the format (sorry I didn't run clang-format on this code :) ):

```
::testing::Values(CatalogClientTestConfiguration{1000, /* requests_per_second */
10,      /* calling_thread_count */
20,     /* task_scheduler_capacity  */
3600  /* runtime */},
```